packages/components/src/components/input/input.component.ts

```diff
@@ -0,0 +1,423 @@
+
+      this.updateComplete.then(() => {
+        if (this.inputElement) {
+          this.inputElement.checkValidity();
+          this.setValidityFromInput();
+          this.internals.setFormValue(this.inputElement.value);
+        }
+      }).catch((error) => {
+        if (this.onerror) {
+          this.onerror(error);
+        }
+      });
+    }
+
+  /**
+   * @internal
+   */
+  @state() prevHelperText = '';
+
+  /**
+   * @internal
+   */
+  @state() prevHelperTextType: ValidationType = DEFAULTS.VALIDATION;
+
+  protected override updated(changedProperties: PropertyValueMap<any> | Map<PropertyKey, unknown>): void {
+    super.updated(changedProperties);
+    if (changedProperties.has('value')) {
+      this.handleValueChange();
+    }
+  }
+
+  /**
+   * Handles the value change of the input field.
+   * Sets the form value and updates the validity of the input field.
+   * @returns void
+   */
+  handleValueChange() {
+    this.internals.setFormValue(this.value);
+    this.updateComplete.then(() => {
+      this.setValidityFromInput();
+    }).catch((error) => {
+      if (this.onerror) {
+        this.onerror(error);
+      }
+    });
+  }
+
+  /**
+   * This function is called when the attribute changes.
+   * It updates the validity of the input field based on the input field's validity.
+   *
+   * @param name - attribute name
+   * @param old - old value
+   * @param value - new value
+   */
+  override attributeChangedCallback(
+    name: string,
+    old: string | null,
+    value: string | null,
+  ): void {
+    super.attributeChangedCallback(name, old, value);
+
+    const validationRelatedAttributes = [
+      'maxlength',
+      'minlength',
+      'pattern',
+      'required',
+    ];
+
+    if (validationRelatedAttributes.includes(name)) {
+      this.updateComplete.then(() => {
+        this.setValidityFromInput();
+      }).catch((error) => {
+        if (this.onerror) {
+          this.onerror(error);
+        }
+      });
+    }
+  }
+
+  /**
+   * Sets the validity of the input field based on the input field's validity.
+   * @returns void
+   */
+  private setValidityFromInput() {
+    if (this.inputElement) {
+      this.internals.setValidity(
+        this.inputElement.validity,
+        this.inputElement.validationMessage,
+        this.inputElement,
+      );
+    }
+  }
+
+  /**
+   * Updates the value of the input field.
+   * Sets the form value.
+   * @returns void
+   */
+  private updateValue() {
+    this.value = this.inputElement.value;
+    this.internals.setFormValue(this.inputElement.value);
+  }
+
+  /**
+   * Handles the input event of the input field.
+   * Updates the value and sets the validity of the input field.
+   *
+   */
+  private onInput() {
+    this.updateValue();
+    this.setValidityFromInput();
+  }
+
+  /**
+   * Handles the change event of the input field.
+   * Updates the value and sets the validity of the input field.
+   * Dispatches the change event.
+   *
+   * @param event
```

Comment:
To avoid the tslint, error we need to add description about the passed argument parameter.
```suggestion
   * @param event - Event which contains information about the value change.
```